### ISSUE TITLE
DatePickerAndroid can select date after maxDate

### ISSUE THREAD

**Issue by tbergq** [facebook/react-native#25189]
React Native version:
React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
      Memory: 6.52 GB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.4.0 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5014246
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3 
      react-native: 0.59.2 => 0.59.2 (also tested on 0.59.9)
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7

## Steps To Reproduce
1. Set start date = today + 36 days.
2. Open AndroidDatepicker with maxDate = today + 365 days
3. Click on year and select next year
4. Click on save and see that date is higher than max date 

Describe what you expected to happen:
It should not be possible to click ok while date is higher than max date, or set date = maxDate. 

Snack, code example, or link to a repository:

```
function addDays(date: Date, days: number): Date {
  const result = new Date(date);
  result.setDate(result.getDate() + days);
  return result;
}
DatePickerAndroid.open({
        date: addDays(new Date(), 36),
        maxDate: addDays(new Date(), 365),
        mode: 'default',
      })
```
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/7705041/59102538-b03f1000-892c-11e9-95be-8e9c918ee340.gif)

Note that max date is set to be 6th of June 2020. After selecting year 2020 and clicking ok, 13th of July is selected, which should not be possible.

**Comment by yeswanth** [comment]
I was able to reproduce this bug on Android 8.0 phone.

I tried to debug this and noted few observations
When `date` and `maxDate` are set
1) Once I move to the next year, Android automatically selects the `date` of the next year. This is shown in the title. This seems to be the normal behaviour
2) Above behaviour also occurs, If the day of `date` current is greater than the day of `maxDate` (without taking into consideration of maxDate value). Hence the bug. 

While going through `react-native` repository, I didn't find any leads on where the error could lie. The implementation of `DatePickerAndroid` in `RNAndroid` project is just a wrapper built on top of existing Android APIs, namely, `DatePickerDialog` and `DatePicker` and they seem fine. 

Will try creating a native Android example to see if I can reproduce it there.
